APE_Black_Formatting.rst

```diff
@@ -0,0 +1,307 @@
+.. _Black: https://github.com/psf/black
+.. _Django DEP-0008: https://github.com/django/deps/blob/main/final/0008-black.rst
+.. _pandas pre-commit: https://pandas.pydata.org/docs/development/contributing_codebase.html#pre-commit
+.. _coding style: https://docs.astropy.org/en/latest/development/codeguide.html#coding-style-conventions
+
+Formatting Code with Black
+===========================
+
+authors: Tom Aldcroft, Erik Tollerud
+
+date-created: 2022 July 13
+
+date-last-revised: 2022 August 25
+
+date-accepted: <replace with accepted date>
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+This APE proposes that the `astropy core package
+<https://github.com/astropy/astropy>`_ adopt Black_ for formatting code.
+
+Detailed description
+--------------------
+
+Foreword
+^^^^^^^^^
+
+This APE presents a rationale that code formatting with Black_ can be beneficial
+both for the ``astropy`` code development process and for the community.
+Hereafter we refer to the astropy core package as ``astropy``, noting that it is
+distinct from the Astropy project as a whole and coordinated and affiliated
+Astropy packages. While coordinated or affiliated packages may choose to adopt
+this APE's conventions in the future, this APE is concerned solely with the core
+package itself.
+
+Black_ is a fast and reliable Python automated code formatter supported by the
+Python Software Foundation. The project motto "The uncompromising code
+formatter" reflects that the code is strongly opinionated about formatting and
+there are very few configuration options. From their documentation, "By using
+*Black*, you agree to cede control over minutiae of hand-formatting. In return,
+Black gives you speed, determinism, and freedom from ``pycodestyle`` nagging
+about formatting."
+
+Code formatting is controversial and the `astropy-dev thread`_ about Black
+generated a spirited discussion but did not result in a clear consensus. This
+strong division of opinions about automated code formatting is common and has
+been documented in other projects, for instance in `Django DEP-0008`_.
+
+Other projects, including NumPy and SciPy, have considered using Black for
+formatting (see e.g. `SciPy PR #14330
+<https://github.com/scipy/scipy/pull/14330>`_). Neither of those projects felt
+that auto-formatting with the current version of Black (circa 2022) was a good
+choice for their codebase, with particular emphasis on needing improvements
+in formatting of mathematical expressions.
```

Comment:
It is worth pointing out that the SciPy issue mentions psf/black#538, which is resolved by now, so not all of the SciPy discussion is relevant anymore.